src/System.IO.FileSystem/tests/FileStream/ctor_str_fm_fa_fs_buffer_fo.cs

```diff
@@ -47,8 +47,21 @@ public void ValidFileOptions(FileOptions option)
             byte[] data = new byte[c_DefaultBufferSize];
             new Random(1).NextBytes(data);
 
-            using (FileStream fs = CreateFileStream(GetTestFilePath(), FileMode.Create, FileAccess.ReadWrite, FileShare.Read, c_DefaultBufferSize, option))
+            FileStream fs = null;
+            try
             {
+                try
+                {
+                    fs = CreateFileStream(GetTestFilePath(), FileMode.Create, FileAccess.ReadWrite, FileShare.Read, c_DefaultBufferSize, option);
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    if ((option & FileOptions.Encrypted) == FileOptions.Encrypted)
```

Comment:
Rather than modify this existing test, it would be cleaner to leave this one as-is but move the `Encrypted` InlineDatas to a separate theory. That way you could avoid having to do this check and wouldn't need to nest try/catches.